File: 프로그래머스/0/120890. 가까운 수/가까운 수.py
def solution(array, n):
    answer = []
    for i in array:
        answer.append(abs(n-i))
    result = []
    for i in array:
        if abs(n-i)==min(answer):
            result.append(i)
    if len(result) >1:
        return min(result)
    else:
        return result[0]

File: 프로그래머스/0/120890. 가까운 수/가까운 수.java
class Solution {
    public int solution(int[] array, int n) {
        int answer = array[0];
        int near_diff = Math.abs(answer - n);
        int diff = 0;
        for (int i : array) {
            diff = Math.abs(i - n);
            if (diff < near_diff) {
                answer = i;
                near_diff = diff;
            } else if (diff == near_diff) {
                if (i < answer) {answer = i;}
            }
        }
        return answer;
    }
}